File: Farhaan/Week-3 Problems/sort-list.cpp
class Solution {
public:
    ListNode* sortList(ListNode* head) {
        
        ListNode *root=NULL;
        
        if(!head)
            return NULL;
        else if(head->next==NULL)
            return head;
        else{
            vector<int> v;
            while(head){
            v.push_back(head->val);
            head=head->next;
            }
            sort(v.begin(), v.end());
            root = new ListNode(v[0]);
            ListNode *t=root;
            for(int i=1; i<v.size(); i++){
            ListNode *r=new ListNode(v[i]);
            t->next=r;
            t=r;
            }
        }
        return root;
    }
};

File: Karthik/19-9_to_20-9/sortlist.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def sortList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        b=[]
        tail=head
        main=head
        while(tail):
            b.append(tail.val)
            tail=tail.next
        b.sort()
        for i in b:
            main.val=i
            main=main.next
        # print(main)
        return head
            